dcos_installer/config.py

```diff
@@ -45,11 +45,11 @@ def normalize_config_validation(messages):
     return validation
 
 
-def make_default_config_if_needed(config_path):
+def make_default_config_if_needed(config_path, config_extra=""):
```

Comment:
This muddles the function's purpose -- it's no longer default config if you're adding to it, and if you're passing `config_extra`, you'll probably not want this function to return without writing it out. Looking at the call site, looks like you can just build your config and call `write_string()` on it directly.